File: rocket-bi-web/di_core/DataIngestion/Domain/Response/GoogleServiceAccountSource.ts
import { DataSourceType, DataSources } from '@core/DataIngestion';
import { DataSource } from '@core/DataIngestion/Domain/Response/DataSource';
import { SourceId } from '@core/domain';

export class GoogleServiceAccountSource implements DataSource {
  id: SourceId;
  orgId: string;
  displayName: string;
  credential: string;
  readonly className = DataSources.GoogleServiceAccountSource;
  lastModify: number;

  constructor(id: SourceId, orgId: string, displayName: string, credential: string, lastModify: number) {
    this.id = id;
    this.orgId = orgId;
    this.displayName = displayName;
    this.credential = credential;
    this.lastModify = lastModify;
  }

  static fromObject(obj: any): GoogleServiceAccountSource {
    return new GoogleServiceAccountSource(obj.id, obj.orgId, obj.displayName, obj.credential, obj.lastModify);
  }
}

File: rocket-bi-server/src/main/scala/co/datainsider/jobscheduler/domain/source/GoogleServiceAccountSource.scala
package co.datainsider.jobscheduler.domain.source

import co.datainsider.jobscheduler.domain.Ids.SourceId
import co.datainsider.jobscheduler.domain.source.DataSourceType.DataSourceType

case class GoogleServiceAccountSource(
    orgId: Long,
    id: SourceId,
    displayName: String,
    credential: String,
    creatorId: String = "",
    lastModify: Long = System.currentTimeMillis()
) extends DataSource {
  override def getId: SourceId = id

  override def getName: String = displayName

  override def getType: DataSourceType = DataSourceType.GoogleServiceAccountCredential

  override def getConfig: Map[String, Any] =
    Map(
      "credential" -> credential
    )

  override def getCreatorId: String = creatorId

  override def getLastModify: SourceId = lastModify
}
